ast/annotations.go

```diff
@@ -509,6 +509,37 @@ func (a *Annotations) toObject() (*Object, *Error) {
 	return &obj, nil
 }
 
+func attachRuleAnnotations(mod *Module) {
+	// make a copy of the annotations
+	cpy := make([]*Annotations, len(mod.Annotations))
+	for i, a := range mod.Annotations {
+		cpy[i] = a.Copy(a.node)
+	}
+
+	for _, rule := range mod.Rules {
+		var j int
+		var found bool
+		for i, a := range cpy {
+			if rule.Loc().Row > a.Location.Row {
```

Comment:
Since we're only looking at rules declared below the annotation, won't we be dropping document-scope annotations for rules that don't happen to be located below the annotation, but have an applicable path?

E.g. this policy might not be good form, but the annotation block applies to both rules `data.example.p`:

```
package example

import rego.v1

p contains {1, rego.metadata.chain()}

# METADATA
# scope: document
# title: p-rules
p contains {2, rego.metadata.chain()}
```